apps/api/src/admin/applicant_review_processor.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any
+
+from pydantic import EmailStr
+
+from models.ApplicationData import Decision
+
+
+def _extract_personalizations(decision_data: dict[str, Any]) -> tuple[str, EmailStr]:
+    name = decision_data["first_name"]
+    email = _recover_email_from_uid(decision_data["_id"])
+    return name, email
+
+
+def _recover_email_from_uid(uid: str) -> str:
+    """For NativeUsers, the email should still delivery properly."""
+    uid = uid.replace("..", "\n")
+    *reversed_domain, local = uid.split(".")
+    local = local.replace("\n", ".")
+    domain = ".".join(reversed(reversed_domain))
+    return f"{local}@{domain}"
+
+
+def _include_review_decision(applicant_record: dict[str, Any]) -> None:
+    """Sets the applicant's decision as the last submitted review decision or None."""
+    reviews = applicant_record["application_data"]["reviews"]
+    applicant_record["decision"] = reviews[-1][2] if reviews else None
+
+
+def _include_num_reviewers(applicant_record: dict[str, Any]) -> None:
+    applicant_record["num_reviewers"] = _get_num_unique_reviewers(applicant_record)
+
+
+def _include_avg_score(applicant_record: dict[str, Any]) -> None:
+    applicant_record["avg_score"] = _get_avg_score(
+        applicant_record["application_data"]["reviews"]
+    )
+
+
+def _delete_reviews(applicant_record: dict[str, Any]) -> None:
+    del applicant_record["application_data"]["reviews"]
+
+
+def _get_last_score(reviewer: str, reviews: list[tuple[str, str, float]]) -> float:
+    for i in range(len(reviews) - 1, -1, -1):
+        if reviews[i][1] == reviewer:
+            return reviews[i][2]
```

Comment:
Suggestion: using an iterable directly would be simpler than using an index, e.g.
```python
for review in reversed(reviews):
```